src/smt/command.h

```diff
@@ -804,6 +808,27 @@ class CVC4_PUBLIC GetOptionCommand : public Command {
   std::string getCommandName() const throw();
 };/* class GetOptionCommand */
 
+// Set expression name command
+// Note this is not an official smt2 command
+// Conceptually:
+//   (assert (! expr :named name))
+// is converted to
+//   (assert expr)
+//   (set-expr-name expr name)
+class CVC4_PUBLIC SetExpressionNameCommand : public Command {
+protected:
+  Expr d_expr;
+  std::string d_name;
+public:
+  SetExpressionNameCommand(Expr expr, std::string name) throw();
```

Comment:
`throw()` is deprecated, use `noexcept` instead.

Reply:
I've changed this in my latest commit.